File: Medium/Java/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int water = 0;
        int left = 0;
        int right = height.length - 1;

        while(left < right){
            int min = Math.min(height[left], height[right]);
            water = Math.max(water, min * (right - left));
            if(height[left] < height[right])
                left++;
            else
                right--;
        }

        return water;
    }
}

File: Medium/JavaScript/11. Container With Most Water.js
var maxArea = function(height) {
    let water = 0;
    let left = 0;
    let right = height.length - 1;

    while(left < right){
        let min = Math.min(height[left], height[right]);
        water = Math.max(water, min * (right - left));
        if(height[left] < height[right])
            left++;
        else
            right--;
    }

    return water;
};
